internal/appsec/dyngo/instrumentation/httpinstr/http.go

```diff
@@ -60,64 +60,92 @@ func init() {
-			}
-		}
-		host := r.Host
-		headers["host"] = []string{host}
 		op := StartOperation(
-			HandlerOperationArgs{
-				Span:       span,
-				IsTLS:      r.TLS != nil,
-				Method:     r.Method,
-				Host:       r.Host,
-				Path:       r.URL.Path,
-				RequestURI: r.RequestURI,
-				RemoteAddr: r.RemoteAddr,
-				Headers:    headers,
-				Cookies:    cookies,
-				// TODO(julio): avoid actively parsing the query string and move to a lazy monitoring of this value with
-				//   the dynamic instrumentation of the Query() method.
-				Query: r.URL.Query(),
-			},
+			makeHandlerOperationArgs(r, span),
 			nil,
 		)
 		defer func() {
 			var status int
 			if mw, ok := w.(interface{ Status() int }); ok {
 				status = mw.Status()
 			}
-			op.Finish(HandlerOperationRes{Status: status})
+			op.Finish(makeHandlerOperationRes(status))
 		}()
 		handler.ServeHTTP(w, r)
 	})
 }
 
-// TODO(julio): create a go-generate tool to generate the types, vars and methods below
+// MakeHandlerOperationArgs creates the HandlerOperationArgs out of a standard
+// http.Request along with the given current span. It returns an empty structure
+// when appsec is disabled.
+func MakeHandlerOperationArgs(r *http.Request, span ddtrace.Span) HandlerOperationArgs {
+	if !enabled {
+		return HandlerOperationArgs{}
+	}
+	return makeHandlerOperationArgs(r, span)
+}
+
+// makeHandlerOperationArgs implements MakeHandlerOperationArgs regardless of appsec being disabled.
+func makeHandlerOperationArgs(r *http.Request, span ddtrace.Span) HandlerOperationArgs {
+	headers := make(http.Header, len(r.Header))
+	for k, v := range r.Header {
+		k := strings.ToLower(k)
+		if k == "cookie" {
+			// Do not include cookies in the request headers
+			continue
+		}
+		headers[k] = v
+	}
+	var cookies map[string][]string
+	if reqCookies := r.Cookies(); len(reqCookies) > 0 {
+		cookies = make(map[string][]string, len(reqCookies))
+		for _, cookie := range reqCookies {
+			if cookie == nil {
+				continue
+			}
+			cookies[cookie.Name] = append(cookies[cookie.Name], cookie.Value)
+		}
+	}
+	headers["host"] = []string{r.Host}
+	return HandlerOperationArgs{
+		Span:       span,
+		IsTLS:      r.TLS != nil,
+		Method:     r.Method,
+		Host:       r.Host,
+		Path:       r.URL.Path,
+		RequestURI: r.RequestURI,
+		RemoteAddr: r.RemoteAddr,
+		Headers:    headers,
+		Cookies:    cookies,
+		// TODO(Julio-Guerra): avoid actively parsing the query string and move to a lazy monitoring of this value with
+		//   the dynamic instrumentation of the Query() method.
+		Query: r.URL.Query(),
+	}
+}
+
+// MakeHandlerOperationRes creates the HandlerOperationRes structure.
+func MakeHandlerOperationRes(status int) HandlerOperationRes {
+	if !enabled {
+		return HandlerOperationRes{}
+	}
+	return makeHandlerOperationRes(status)
+}
+
+// makeHandlerOperationRes implements MakeHandlerOperationRes regardless of appsec being disabled.
+func makeHandlerOperationRes(status int) HandlerOperationRes {
```

Comment:
This function just adds another layer of complexity. Just writing `HandlerOperationRes{Status: status}` everywhere should be equally good and much clearer.

Reply:
Reverted, now that we no longer need to no-op it.